src/shared/sam/cli/samCliContext.ts

```diff
@@ -53,6 +55,26 @@ export async function getSamCliVersion(context: SamCliContext): Promise<string>
     return result.versionValidation!.version!
 }
 
+/**
+ * Returns the correct Docker image name based on SAM CLI version.
+ * Anything not greater than 1.0.0, undefined, or non-parseable will use `amazon/aws-sam-cli-emulation-image-${lambdaRuntime}`
+ * Else, use `lambci/lambda:${lambdaRuntime}`
+ * @param samCliVersion Version to evaluate
+ * @param lambdaRuntime Runtime name to append to Docker image
+ */
+export function getSamCliDockerImageNameWithRuntime(samCliVersion: string | undefined, lambdaRuntime: string): string {
```

Comment:
nit: "withRuntime" in the function name is redundant because `lambdaRuntime` is already a parameter. Such a function name implies that there might be other `getSamCliDockerImageNameXX` functions (a "family") to disambiguate. But that's unlikely, so the verbose name can be avoided.